### ISSUE TITLE
bvaio pop_writebva error: Unrecognized field name "nbchan"

### ISSUE THREAD

**Issue by Valerie123hi** [arnodelorme/bva-io#20]
Hello!

I have a problem with exporting EEG data from Eeglab to Brainvision Analyzer (via bvaio).

It works fine to load existing datasets (also the whole database).
However, I can only export one file at a time (manually by clicking on file - - > export - - > Write Brain Vis. exchange format file in Eeglab)

I copied the history file of this operation and tried to apply it to all files:

EEG.etc.eeglabvers = '2021.1'; % this tracks which version of EEGLAB is being used, you may ignore it
EEG = eeg_checkset( EEG );
EEG = eeg_checkset( EEG );
EEG = eeg_checkset( EEG );
EEG = eeg_checkset( EEG );
EEG = eeg_checkset( EEG );
EEG = eeg_checkset( EEG );
EEG.etc.eeglabvers = '2022.0'; % this tracks which version of EEGLAB is being used, you may ignore it
EEG = eeg_checkset( EEG );
pop_writebva(EEG,'E:\folder\allfiles(datafile).name');

I always get this error:

Unrecognized field name "nbchan".

Error in pop_writebva (line 94)
for index = 1:EEG.nbchan

Line 94 looks like this (it's the 3rd line here)

% write data
% ----------
for index = 1:EEG.nbchan
fwrite(fid3, EEG.data(index,:), 'float' );
end;

What could be the problem, and how can I solve it other than having to export all 580 files separately?

Thank you!
Valerie

**Comment by arnodelorme** [comment]
Yes, you can write a loop to export the data files. The issue is likely that your dataset is empty.

**Comment by Valerie123hi** [comment]
Hello, thank you for your reply!

The dataset is not empty, but it seems like there was a problem trying to load all files at once, so important information like the number of channels was not displayed correctly. Another problem was that I used a very old Matlab version - it seems like i got a different output after updating the program.

I was still not able to write a correct loop but found a different solution. I exported all files using the commands from the history file (after loading and exporting one file manually) and adapting the file names individually (was quite easy with word, find & replace the respective values to include subject 1 to 580).
Maybe this is helpful for someone else if they come across the same problem :)
